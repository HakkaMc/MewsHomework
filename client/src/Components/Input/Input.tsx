import React from "react";
import {
	Field,
	WrappedFieldProps,
	WrappedFieldInputProps,
	CommonFieldProps
} from "redux-form";
import { TextField, InputBaseProps } from "@material-ui/core";

const Component: React.FC<WrappedFieldInputProps &
	InputBaseProps &
	CommonFieldProps &
	WrappedFieldProps> = (
	props: WrappedFieldInputProps &
		InputBaseProps &
		CommonFieldProps &
		WrappedFieldProps
) => {
	const updatedProps: any = {
		...props,
		...props.input
	};

	delete updatedProps.meta;

	return <TextField {...updatedProps} />;
};

const Input: React.FC<WrappedFieldInputProps &
	InputBaseProps &
	CommonFieldProps &
	any> = props => <Field {...props} component={Component} />;

export default Input;
