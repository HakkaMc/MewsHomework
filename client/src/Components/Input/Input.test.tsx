import React from "react";
import { shallow } from "enzyme";

import Input from "./Input";

describe("<Input/>", () => {
	it("Match snapshot", () => {
		const wrapper = shallow(<Input />);
		expect(wrapper).toMatchSnapshot();
	});
});
