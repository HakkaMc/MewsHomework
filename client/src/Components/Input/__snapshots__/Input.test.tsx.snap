// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Input/> Match snapshot 1`] = `
<Field
  component={[Function]}
/>
`;
